app/controllers/doorkeeper_applications_controller.rb

```diff
@@ -17,6 +17,11 @@ def update
     end
   end
 
+  def users_with_access
+    authorized_users_user_ids = @application.authorized_tokens.select(:resource_owner_id)
+    @users = policy_scope(User).where(id: authorized_users_user_ids).includes(:organisation).page(params[:page]).per(100)
```

Comment:
Good to know! 

That won't be a straightforward query probably? It looks like I have to join `users`, `user_application_permissions` and `supported_permissions`. Am I on the right track?